libs/execution_context/ebpf_maps.c

```diff
@@ -188,8 +188,15 @@ _ebpf_core_circular_map_peek_or_pop(_Inout_ ebpf_core_circular_map_t* map, bool
             map->end = new_end;
         }
     }
+    if (pop) {
+        // The return_value is not freed until the current epoch is retired.
+        ebpf_epoch_free(return_value);
+    }
 Done:
+#pragma warning(push)
+#pragma warning(disable : 6001) // Using uninitialized memory 'return_value'.
```

Comment:
I'd suggest adding `_Ret_maybenull_` to line 162 but that can be in a separate PR